internal/controller/api/v1alpha1/host_types.go

```diff
@@ -17,25 +17,180 @@ limitations under the License.
+}
+
+type HostIoPerf struct {
+	// 99th percentile of fsync duration in milliseconds
+	SyncDuration int64 `json:"syncDuration,omitempty"`
+}
+
+type HostDisk struct {
+	DriveType               string                      `json:"driveType,omitempty"`
+	Vendor                  string                      `json:"vendor,omitempty"`
+	Name                    string                      `json:"name,omitempty"`
+	Path                    string                      `json:"path,omitempty"`
+	Hctl                    string                      `json:"hctl,omitempty"`
+	ByPath                  string                      `json:"byPath,omitempty"`
+	Model                   string                      `json:"model,omitempty"`
+	Wwn                     string                      `json:"wwn,omitempty"`
+	Serial                  string                      `json:"serial,omitempty"`
+	SizeBytes               int64                       `json:"sizeBytes,omitempty"`
+	Bootable                bool                        `json:"bootable,omitempty"`
+	Smart                   string                      `json:"smart,omitempty"`
+	InstallationEligibility HostInstallationEligibility `json:"installationEligibility,omitempty"`
+	IoPerf                  HostIoPerf                  `json:"ioPerf,omitempty"`
+}
+
+type HostBoot struct {
+	CurrentBootMode string `json:"currentBootMode,omitempty"`
+	PxeInterface    string `json:"pxeInterface,omitempty"`
+}
+
+type HostSystemVendor struct {
+	SerialNumber string `json:"serialNumber,omitempty"`
+	ProductName  string `json:"productName,omitempty"`
+	Manufacturer string `json:"manufacturer,omitempty"`
+	Virtual      bool   `json:"virtual,omitempty"`
+}
+
+type HostInventory struct {
+	Timestamp    *metav1.Time     `json:"timestamp,omitempty"`
+	Hostname     string           `json:"hostname,omitempty"`
+	BmcAddress   string           `json:"bmcAddress,omitempty"`
+	BmcV6address string           `json:"bmcV6address,omitempty"`
+	Memory       HostMemory       `json:"memory,omitempty"`
+	CPU          HostCPU          `json:"cpu,omitempty"`
+	Interfaces   []HostInterface  `json:"interfaces"`
+	Disks        []HostDisk       `json:"disks"`
+	Boot         HostBoot         `json:"boot,omitempty"`
+	SystemVendor HostSystemVendor `json:"systemVendor,omitempty"`
+}
 
 // HostSpec defines the desired state of Host
 type HostSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// The name of the Cluster CR
+	ClusterRef              *corev1.ObjectReference `json:"clusterRef"`
+	Role                    models.HostRole         `json:"role" protobuf:"bytes,1,opt,name=role,casttype=HostRole"`
+	Hostname                string                  `json:"hostname,omitempty"`
+	MachineConfigPool       string                  `json:"machineConfigPool,omitempty"`
+	Enabled                 bool                    `json:"enabled,omitempty"`
```

Comment:
Not sure `omitempty` makes sense here unless you make it a `*bool` to represent "unset". There are other cases of this below.

Reply:
Changed to *bool